data_structures/src/transaction.rs

```diff
@@ -466,14 +466,53 @@ pub struct MintTransaction {
 
 impl MintTransaction {
     /// Creates a new mint transaction.
-    pub fn new(epoch: Epoch, output: ValueTransferOutput) -> Self {
+    pub fn new(epoch: Epoch, outputs: Vec<ValueTransferOutput>) -> Self {
         MintTransaction {
             epoch,
-            output,
+            outputs,
             hash: MemoHash::new(),
         }
     }
 
+    #[allow(clippy::too_many_arguments)]
+    pub fn with_split_utxos(
+        epoch: Epoch,
+        reward: u64,
+        own_pkh: PublicKeyHash,
+        own_utxos: &HashMap<OutputPointer, u64>,
+        unspent_outputs_pool: &UnspentOutputsPool,
+        collateral_minimum: u64,
+        collateral_age: u32,
+        data_request_max_retrievals_per_epoch: u16,
+    ) -> Self {
+        let mut vt_outputs = vec![];
+        let mut utxo_counter =
```

Comment:
Actually, this should also be an argument calculated externally, so the final list of arguments should be:


    pub fn with_split_utxos(
        epoch: Epoch,
        reward: u64,
        own_pkh: PublicKeyHash,
        collateral_minimum: u64,
        utxos_to_achieve: usize,
    ) -> Self {